### PR TITLE
refactor: user components

### LINKED ISSUE #1257: [REFACTOR]: User Components
## Summary
Currently, the design of our user components is inconsistent and does not follow a standardised pattern. This makes it difficult for developers to maintain and update the codebase, as they have to constantly adapt to new design patterns and layouts.

To address this issue, we need to refactor the design of every user component in our application to create a more cohesive and consistent experience for developers. As well as document those components on our Storybook.

This includes:
 - Standardizing color palettes, typography, and layout across all user components to create a consistent design system
 - Updating icons and buttons to create reusable components that can be easily customized and extended
 - Simplifying and decluttering the user interface to reduce the complexity of the codebase

By refactoring the design of every user component, we can improve developer efficiency and reduce the time required to implement new features and fix bugs. This can result in a faster development cycle, improved code quality, and ultimately a better user experience.

### PR DESCRIPTION
**Pull request by StereoPT**
Relates to:
- #1257 

Fixes:
- #1257

## Proposed Changes

  - Refactored the components used in the User Pages

This pull request closes #1257

### CODE REVIEW

**Comment by JoaoSaIvador** [comment]
This pull request also closes the issue #1119

File: frontend/src/components/Primitives/Badge/Badge.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        lineHeight: '$20',
+      },
+      xs: {
+        fontSize: '$12',
+        lineHeight: '$16',
+      },
+    },
+    pill: {
+      true: {
+        borderRadius: '50rem',
```
Comment (nunocaseiro) [inline frontend/src/components/Primitives/Badge/Badge.tsx:54]:
```suggestion
        borderRadius: '$800',
```

I'm not sure about this but 50 * 16 = 800

File: frontend/src/components/Users/User/Header/Header.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    // Act
+    const { getByTestId } = render({ user });
+    const title = getByTestId('userHeader').querySelector('span');
+
+    // Assert
+    expect(title).toHaveTextContent(`${user.firstName} ${user.lastName}`);
+  });
+
+  it('should render breadcrumbs correctly', async () => {
+    // Arrange
```
Comment (nunocaseiro) [inline frontend/src/components/Users/User/Header/Header.spec.tsx:29]:
```suggestion
```

File: frontend/src/components/Users/UsersList/UserItem/UserItemActions/UserItemActions.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    try {
+      await mutateAsync(updateTeamUser);
+      setCheckedState(checked);
+    } catch (error) {
+      setCheckedState(!checked);
+    }
+  };
+
+  const {
+    deleteUser: { mutate },
```
Comment (nunocaseiro) [inline frontend/src/components/Users/UsersList/UserItem/UserItemActions/UserItemActions.tsx:43]:
```suggestion
    deleteUser: { mutate: deleteUserMutation },
```

File: frontend/src/components/Users/UsersList/UserItem/UserItemActions/UserItemActions.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+type UserItemActionsProps = {
+  user: User;
+};
+
+const UserItemActions = React.memo(({ user }: UserItemActionsProps) => {
+  const { userId } = useCurrentSession();
+  const [checkedState, setCheckedState] = useState(user.isSAdmin);
+
+  const {
+    updateUserIsAdmin: { mutateAsync },
```
Comment (nunocaseiro) [inline frontend/src/components/Users/UsersList/UserItem/UserItemActions/UserItemActions.tsx:25]:
```suggestion
    updateUserIsAdmin: { updateUserMutation },
```

File: frontend/src/components/Users/UsersList/UserItem/UserItemActions/UserItemActions.tsx [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+  } = useUser();
+
+  const handleSuperAdminChange = async (checked: boolean) => {
+    const updateTeamUser: UpdateUserIsAdmin = {
+      _id: user._id,
+      isSAdmin: checked,
+    };
+
+    try {
+      await mutateAsync(updateTeamUser);
```
Comment (nunocaseiro) [inline frontend/src/components/Users/UsersList/UserItem/UserItemActions/UserItemActions.tsx:35]:
```suggestion
      await updateUserMutation(updateTeamUser);
```

File: frontend/src/components/Users/UsersList/UserItem/UserItemActions/UserItemActions.tsx [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+      Do you really want to delete the user{' '}
+      <Text fontWeight="bold">
+        {user.firstName} {user.lastName}
+      </Text>
+      ?
+    </Text>
+  );
+
+  const handleDeleteUser = () => {
+    mutate({ id: user._id });
```
Comment (nunocaseiro) [inline frontend/src/components/Users/UsersList/UserItem/UserItemActions/UserItemActions.tsx:57]:
```suggestion
    deleteUserMutation({ id: user._id });
```

**Review by nunocaseiro — CHANGES_REQUESTED** [review]

**Review by nunocaseiro — APPROVED** [review]

**Pull request merged**